tests/test_csc.py

```diff
@@ -0,0 +1,136 @@
+import asyncio
+import unittest
+import logging
+
+from lsst.ts import salobj
+from lsst.ts import ATHexapod
+
+STD_TIMEOUT = 15
+SHORT_TIMEOUT = 5
+
+
+class Harness:
+    def __init__(self, initial_state):
+        salobj.test_utils.set_random_lsst_dds_domain()
+        self.csc = ATHexapod.ATHexapodCSC(initial_state=initial_state)
+        self.remote = salobj.Remote(domain=self.csc.domain, name="ATHexapod", index=0)
+        self.server = ATHexapod.MockServer()
+        self.log = logging.getLogger(__name__)
+
+    async def __aenter__(self):
+        await self.server.start()
+        await self.csc.start_task
+        return self
+
+    async def __aexit__(self, *args):
+        await self.csc.close()
+        await self.server.stop()
+
+
+class CscTestCase(unittest.TestCase):

```

Comment:
Please consider using the `asynctest` package. It makes these tests much easier to read and also gives significantly clearer output when a test fails:

* Add `import asynctest` to the imports. It is a 3rd party library so goes after the standard library imports.
* Make your test case inherit from `asynctest.TestCase` instead of `unittest.TestCase`.
* For async test cases use `async def test_xxx(self):` and get rid of the `async def doit()` and  `asyncio.get_event_loop().run_until_complete(doit())` lines (and outdent the real code 4 spaces).

Converting is quick.

You can look for `asynctest` in `ts_salobj` to see examples, but I doubt you'll need to bother.